src/commands/multi_block/monitor.rs

```diff
@@ -437,9 +638,210 @@ async fn bail(listen: Listen, client: &Client, signer: Signer) -> Result<(), Err
+            rounds_to_remove.push(round_to_check);
+            continue;
+        }
+
+        let maybe_round_phase = storage_finalized
+            .fetch(&runtime::storage().multi_block().current_phase())
+            .await?;
+
+        let mut should_clear = false;
+        let mut clear_reason = "";
+
+        match maybe_round_phase {
+            Some(Phase::SignedValidation(_)) => {
+                let sorted_scores = storage_finalized
+                    .fetch_or_default(
+                        &runtime::storage()
+                            .multi_block_signed()
+                            .sorted_scores(round_to_check),
+                    )
+                    .await?;
+                if let Some((_, winning_score)) = sorted_scores.0.first() {
+                    if score_passes_strategy(
+                        winning_score.0,
+                        submitted_score,
+                        SubmissionStrategy::IfLeading,
+                    ) && winning_score.0 != submitted_score
+                    {
+                        should_clear = true;
+                        clear_reason =
+                            "Better solution validated during its SignedValidation phase";
+                    }
+                }
+            }
+            Some(Phase::Signed(_)) => {}
+            Some(_) | None => {
+                should_clear = true;
+                clear_reason = "Phase is past Signed/SignedValidation or round gone, and metadata still exists";
+            }
+        }
+
+        if should_clear {
+            log::info!(target: LOG_TARGET, "Submission for past round {} detected as DISCARDED (Reason: {}). Attempting to clear.", round_to_check, clear_reason);
+            match clear_submission(
+                config.listen,
+                client,
+                signer.clone(),
+                round_to_check,
+                n_pages,
+            )
+            .await
+            {
+                Ok(_) => {
+                    log::info!(target: LOG_TARGET, "Successfully cleared submission for past round {}", round_to_check);
+                    rounds_to_remove.push(round_to_check);
+                }
+                Err(e) => {
+                    log::error!(target: LOG_TARGET, "Failed to clear submission for past round {}: {:?}. Will retry.", round_to_check, e);
+                }
+            }
+        }
+    }
 
-    utils::wait_tx_in_block_for_strategy(tx, listen).await?;
+    if !rounds_to_remove.is_empty() {
+        for round_index in rounds_to_remove {
+            untrack_round_submission(
+                submitted_rounds,
+                round_index,
+                Some("Removed past round {} from local tracking after clearing"),
+            )
+            .await;
+        }
+    }
 
     Ok(())
 }
+
+async fn collect_past_rounds_to_clear(
+    submitted_rounds: &Arc<Mutex<HashMap<u32, (u32, ElectionScore)>>>,
```

Comment:
deref impl of Arc is just &Mutex no need to for &Arc<Mutex<....>

```suggestion
    submitted_rounds: &Mutex<HashMap<u32, (u32, ElectionScore)>>>,
```